### ISSUE TITLE
Pulse Width Modulation

### ISSUE THREAD

**Issue by SteveDesmond-ca** [ecoAPM/SimpleGPIO#7]
Being able to dim / slow components would be cool. I have no idea how to go about doing it.

**Comment by stevedesmond-ca** [comment]
Gross. Literally just toggling really fast. https://github.com/larsks/raspberry-gpio-python-code/blob/6d62222bd5384eb2dc2ed13065ea28dcb506cc67/source/soft_pwm.c#L90-L114

**Comment by chris-castillo** [comment]
I can work on this if this project is still in development?

**Comment by SteveDesmond-ca** [comment]
The .NET IOT device bindings library has a [software PWM implementation](https://github.com/dotnet/iot/blob/main/src/devices/SoftPwm/SoftwarePwmChannel.cs) that we can wrap.

A quick demo found that the high precision setting is needed for smooth (aka not stepped) transitions and to prevent flickering, but this pins a high priority thread at 100%. That means full RGB LED control would use 75% of a Raspberry Pi's CPU.

60Hz seems acceptable from a visual standpoint, though we may want to double that to 120Hz for those with acute vision, if going with high precision; without high precision, 120Hz flickers.

```
using System.Device.Pwm.Drivers;

var running = true;

var pwm = new SoftwarePwmChannel(26, 1, 0, true);
pwm.Start();

while (running)
{
	var key = Console.ReadKey();
	switch (key.Key)
	{
		case ConsoleKey.UpArrow:
			pwm.DutyCycle += 0.01;
			break;
		case ConsoleKey.DownArrow:
			pwm.DutyCycle -= 0.01;
			break;
		case ConsoleKey.RightArrow:
			pwm.Frequency += 1;
			break;
		case ConsoleKey.LeftArrow:
			pwm.Frequency -= 1;
			break;
		case ConsoleKey.Q:
			running = false;
			break;
	}

	Console.WriteLine($"On: {pwm.DutyCycle:P0} % / Freq: {pwm.Frequency,10} Hz");
}

pwm.Stop();
Console.WriteLine("Done!");
pwm.Dispose();
```

**Comment by SteveDesmond-ca** [comment]
Implemented in `v2.0-beta.2` via `IPinInterface.Strength` -- docs and helpers coming by v2.0 final
